Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -422,3 +566,33 @@ def ReadFile(path):
         LOG_ACTION.log(LogType.Error, "Exception opening file " + path + " Error Code: " + str(error.errno) + " Error: " + error.message + error.strerror)
     return content
 
+def ConvertToUtf8(item):
+    return str(item).encode('utf-8')
+
+def GetWorkspaceListFromStateFile():
+    stateFileFullPath = os.path.join(PATH_NPM_STATE, FNAME_NPM_STATE)
+    fileContents = ReadFile(stateFileFullPath)
+    if fileContents is None:
+        return []
+    wsList = filter(IsUUID, fileContents.splitlines())
+    return map(ConvertToUtf8, wsList)
+
+def IsUUID(uuidStr):
+    try:
+        uuidOut = re.match('[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}\Z', uuidStr, re.I)
```

Comment:
I agree with @lagalbra. The [python uuid](https://docs.python.org/2/library/uuid.html), which generates UUIDs based on [RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) should suffice for your case.